tests/unit/test_testbench.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from numpy.lib.function_base import quantile
+import pytest
+
+import elfi
+import elfi.examples.ma2 as exma2
+from elfi.methods.inference.parameter_inference import ParameterInference
+
+
+def test_testbenchmethod_init():
+
+    method = elfi.TestbenchMethod(method=elfi.SMC, name="SMC_1")
+    method.set_method_kwargs(discrepancy_name='d', batch_size=50)
+    method.set_sample_kwargs(n_samples=100, thresholds=[2.0, 1.0], bar=False)
+    attr = method.get_method()
+
+    assert attr['name'] == "SMC_1"
+    assert attr['method_kwargs']['batch_size'] == 50
+    assert attr['sample_kwargs']['n_samples'] == 100
+
+def test_testbench_init_param_reps(ma2):
+
+    testbench = elfi.Testbench(model=ma2,
+                               repetitions=5,
+                               seed=99,
+                               progress_bar=False)
+
+    for _, values in testbench.reference_parameter.items():
+        assert values.size == 5
+
+
+def test_testbench_init_given_params(ma2):
+
+    ref_params = ma2.generate(batch_size=1, outputs=['t1', 't2'])
+    testbench = elfi.Testbench(model=ma2,
+                               reference_parameter=ref_params,
+                               repetitions=5,
+                               seed=99,
+                               progress_bar=False)
+
+    for _, values in testbench.reference_parameter.items():
+        assert np.all(values == values[0])
+        assert values.size == 5
+
+
+def test_testbench_init_obs_reps(ma2):
+
+    testbench = elfi.Testbench(model=ma2,
+                               repetitions=5,
+                               seed=99,
+                               progress_bar=False)
+
+    assert len(testbench.observations) == 5
+
+
+def test_testbench_init_given_obs(ma2):
+
+    obs = ma2.generate(batch_size=1, outputs=['MA2'])
+    testbench = elfi.Testbench(model=ma2,
+                               observations=obs,
+                               repetitions=5,
+                               seed=99,
+                               progress_bar=False)
+
+    assert len(testbench.observations) == 5
+    assert np.all(
+        [a == b for a, b in zip(testbench.observations, testbench.observations)]
```

Comment:
is this correct or should it be comparison between `testbench.observations` and `obs`